src/parser.rs

```diff
@@ -2040,22 +2042,49 @@ pub fn parse_statement(
             trace!("parsing return");
 
             *index += 1;
-
-            let (expr, err) =
-                parse_expression(tokens, index, ExpressionKind::ExpressionWithoutAssignment);
-            error = error.or(err);
-
-            (
-                ParsedStatement::Return(
-                    expr,
-                    Span {
-                        file_id: start.file_id,
-                        start: start.start,
-                        end: tokens[(*index - 1)].span.end,
-                    },
-                ),
-                error,
-            )
+            if *index < tokens.len() {
+                let tok = &tokens[*index].contents;
+                match tok {
+                    TokenContents::Eol => (
```

Comment:
Don't you have to consume `TokenContents::Eol` token by advancing the index?

Reply:
Yeah. I'll add it.